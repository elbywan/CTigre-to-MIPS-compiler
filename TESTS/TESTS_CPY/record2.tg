type int_list = {hd : int, tl: int_list} in

var x := nil in

x := (int_list {hd=5, tl= int_list{hd=9, tl=x}});

var t := x in
for i=1 to 5 do
  print(">> "); printint(t.hd);print("\n");
  t := t.tl
done