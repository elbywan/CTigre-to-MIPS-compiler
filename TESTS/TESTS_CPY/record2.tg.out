>> 5
>> 9
>> ERROR: null pointer

